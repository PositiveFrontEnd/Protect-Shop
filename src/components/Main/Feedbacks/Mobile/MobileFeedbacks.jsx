import React, { useEffect, useState } from "react";
import { Swiper, SwiperSlide } from "swiper/react";
import "swiper/swiper.min.css";
import "./MobileFeedbacks.scss";
import Button from "../../../Button/Button";
import { useNavigate } from "react-router";
import MobileFeedbackCard from "./mobileCard.jsx";
const MobileFeedbacks = ({ comments }) => {
  const navigate = useNavigate();
  const [randomComments, setRandomComments] = useState([]);
  useEffect(() => {
    const shuffleArray = (array) => {
      const newArray = [...array];
      let currentIndex = newArray.length,
        temporaryValue,
        randomIndex;
      while (0 !== currentIndex) {
        randomIndex = Math.floor(Math.random() * currentIndex);
        currentIndex -= 1;
        temporaryValue = newArray[currentIndex];
        newArray[currentIndex] = newArray[randomIndex];
        newArray[randomIndex] = temporaryValue;
      }
      return newArray;
    };

    const shuffledComments = shuffleArray(comments);

    const selectedComments = shuffledComments.slice(0, 3);
    setRandomComments(selectedComments);
  }, [comments]);
  return (
    <div className="mobile__feedbacks__wrapper">
      <h2 className="feedbacks__title">feedbacks</h2>
      <div>
        <Swiper
          spaceBetween={15}
          slidesPerView={2}
          centeredSlides={true}
          initialSlide={1}
        >
          {randomComments.map((item, index) => (
            <SwiperSlide key={index} className="feedbacks__card">
              <MobileFeedbackCard
                key={index}
                name={item.customer?.firstName || 'Unknown User'}
                avatar={item.customer?.avatarUrl}
                likes={item.likes}
                comment={item.advantages}
                background={item.customer?.background || ''}
              />
            </SwiperSlide>
          ))}
        </Swiper>
      </div>
      <div className="mobile__feedbacks__button">
        <Button click={() => navigate("/feedbacks")} white>
          {" "}
          See More
        </Button>
      </div>
    </div>
  );
};

export default MobileFeedbacks;
