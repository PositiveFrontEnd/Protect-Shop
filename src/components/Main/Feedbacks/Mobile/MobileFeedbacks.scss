.mobile__feedbacks__wrapper {
  padding-block: 24px;
  display: flex;
  flex-direction: column;

  .feedbacks__title {
    margin-bottom: 10px;
    font-size: 18px;
    font-family: "Noto Sans", sans-serif;
    text-align: center;
    text-transform: uppercase;
  }

  .feedbacks__swiper {
    padding-block: 20px;
  }

  .swiper-wrapper {
    display: flex;
    gap: 10px;
  }
  .swiper {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }
  .feedbacks__card {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 12px 8px;
    gap: 12px;
    background-color: var(--feedback);
  }

  .card__image {
    img {
      width: 120px;
      height: 120px;
    }
  }

  .card__stars {
    display: flex;
    gap: 12px;
    height: 20px;
  }
  .star {
    max-width: 16px;
  }

  .card__name {
    margin-top: 32px;
    font-size: 14px;
    text-transform: uppercase;
  }

  .card__description {
    display: flex;
    align-items: start;
    max-height: 120px;
    height: 100%;
    font-size: 12px;
    line-height: 150%;
    overflow: scroll;
  }

  .mobile__feedbacks__button {
    align-self: center;
    width: 100%;
    max-width: 280px;
    margin-top: 16px;
    text-align: center;
  }
}
.star.inactive {
  display: none;
}
