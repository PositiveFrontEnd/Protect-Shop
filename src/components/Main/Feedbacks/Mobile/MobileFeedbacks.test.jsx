import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import MobileFeedbacks from "./MobileFeedbacks";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Тестування компонента  SearchPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore();
        const initialState = {

        };

        const store = mockStore(initialState);
        const mobileFeedbacks = render(
            <Provider store={store}>
                <BrowserRouter>
                    <MobileFeedbacks />
                </BrowserRouter>
            </Provider>
        );

        expect(mobileFeedbacks).toMatchSnapshot();
    });
});
