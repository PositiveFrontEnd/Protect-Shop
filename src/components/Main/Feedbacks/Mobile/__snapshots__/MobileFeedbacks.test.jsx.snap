// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  MobileFeedbacks > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mobile__feedbacks__wrapper"
      >
        <h2
          class="feedbacks__title"
        >
          feedbacks
        </h2>
        <div>
          <div
            class="swiper swiper-initialized swiper-horizontal"
          >
            <div
              class="swiper-wrapper"
              style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
            >
              <div
                class="swiper-slide feedbacks__card"
              >
                <div
                  class="no__avatar__wrapper avatar__mobile"
                >
                  <p
                    class="avatar__letter"
                  >
                    U
                  </p>
                </div>
                <div
                  class="card__stars"
                >
                  <svg
                    class=" star inactive"
                    fill="none"
                    height="20"
                    viewBox="0 0 12 12"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_1704_1143)"
                    >
                      <path
                        d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                        fill="#FDE82A"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_1704_1143"
                      >
                        <rect
                          fill="white"
                          height="12"
                          width="12"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </div>
                <p
                  class="card__name"
                >
                  Unknown User
                </p>
                <p
                  class="card__description"
                />
              </div>
              <div
                class="swiper-slide feedbacks__card"
              >
                <div
                  class="no__avatar__wrapper avatar__mobile"
                >
                  <p
                    class="avatar__letter"
                  >
                    U
                  </p>
                </div>
                <div
                  class="card__stars"
                >
                  <svg
                    class=" star inactive"
                    fill="none"
                    height="20"
                    viewBox="0 0 12 12"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_1704_1143)"
                    >
                      <path
                        d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                        fill="#FDE82A"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_1704_1143"
                      >
                        <rect
                          fill="white"
                          height="12"
                          width="12"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </div>
                <p
                  class="card__name"
                >
                  Unknown User
                </p>
                <p
                  class="card__description"
                />
              </div>
              <div
                class="swiper-slide feedbacks__card"
              >
                <div
                  class="no__avatar__wrapper avatar__mobile"
                >
                  <p
                    class="avatar__letter"
                  >
                    U
                  </p>
                </div>
                <div
                  class="card__stars"
                >
                  <svg
                    class=" star inactive"
                    fill="none"
                    height="20"
                    viewBox="0 0 12 12"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_1704_1143)"
                    >
                      <path
                        d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                        fill="#FDE82A"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_1704_1143"
                      >
                        <rect
                          fill="white"
                          height="12"
                          width="12"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </div>
                <p
                  class="card__name"
                >
                  Unknown User
                </p>
                <p
                  class="card__description"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mobile__feedbacks__button"
        >
          <button
            class="button-white"
            type="button"
          >
             
            See More
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mobile__feedbacks__wrapper"
    >
      <h2
        class="feedbacks__title"
      >
        feedbacks
      </h2>
      <div>
        <div
          class="swiper swiper-initialized swiper-horizontal"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide feedbacks__card"
            >
              <div
                class="no__avatar__wrapper avatar__mobile"
              >
                <p
                  class="avatar__letter"
                >
                  U
                </p>
              </div>
              <div
                class="card__stars"
              >
                <svg
                  class=" star inactive"
                  fill="none"
                  height="20"
                  viewBox="0 0 12 12"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_1704_1143)"
                  >
                    <path
                      d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                      fill="#FDE82A"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_1704_1143"
                    >
                      <rect
                        fill="white"
                        height="12"
                        width="12"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
              <p
                class="card__name"
              >
                Unknown User
              </p>
              <p
                class="card__description"
              />
            </div>
            <div
              class="swiper-slide feedbacks__card"
            >
              <div
                class="no__avatar__wrapper avatar__mobile"
              >
                <p
                  class="avatar__letter"
                >
                  U
                </p>
              </div>
              <div
                class="card__stars"
              >
                <svg
                  class=" star inactive"
                  fill="none"
                  height="20"
                  viewBox="0 0 12 12"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_1704_1143)"
                  >
                    <path
                      d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                      fill="#FDE82A"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_1704_1143"
                    >
                      <rect
                        fill="white"
                        height="12"
                        width="12"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
              <p
                class="card__name"
              >
                Unknown User
              </p>
              <p
                class="card__description"
              />
            </div>
            <div
              class="swiper-slide feedbacks__card"
            >
              <div
                class="no__avatar__wrapper avatar__mobile"
              >
                <p
                  class="avatar__letter"
                >
                  U
                </p>
              </div>
              <div
                class="card__stars"
              >
                <svg
                  class=" star inactive"
                  fill="none"
                  height="20"
                  viewBox="0 0 12 12"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_1704_1143)"
                  >
                    <path
                      d="M6.20634 0.5L7.55342 4.6459H11.9127L8.38597 7.2082L9.73305 11.3541L6.20634 8.7918L2.67963 11.3541L4.02671 7.2082L0.5 4.6459H4.85925L6.20634 0.5Z"
                      fill="#FDE82A"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_1704_1143"
                    >
                      <rect
                        fill="white"
                        height="12"
                        width="12"
                      />
                    </clippath>
                  </defs>
                </svg>
              </div>
              <p
                class="card__name"
              >
                Unknown User
              </p>
              <p
                class="card__description"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="mobile__feedbacks__button"
      >
        <button
          class="button-white"
          type="button"
        >
           
          See More
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
